LPFilter/Source/PluginProcessor.cpp

```diff
@@ -1,188 +1,183 @@
+{
+   #if JucePlugin_IsMidiEffect
+    return true;
+   #else
+    return false;
+   #endif
+}
+
+double LpfilterAudioProcessor::getTailLengthSeconds() const
+{
+    return 0.0;
+}
+
+int LpfilterAudioProcessor::getNumPrograms()
+{
+    return 1;   // NB: some hosts don't cope very well if you tell them there are 0 programs,
+                // so this should be at least 1, even if you're not really implementing programs.
+}
+
+int LpfilterAudioProcessor::getCurrentProgram()
+{
+    return 0;
+}
+
+void LpfilterAudioProcessor::setCurrentProgram (int index)
+{
+}
+
+const String LpfilterAudioProcessor::getProgramName (int index)
+{
+    return {};
+}
+
+void LpfilterAudioProcessor::changeProgramName (int index, const String& newName)
+{
+}
+
+//==============================================================================
+void LpfilterAudioProcessor::prepareToPlay (double sampleRate, int samplesPerBlock)
+{
+    // Use this method as the place to do any pre-playback
+    // initialisation that you need..
+}
+
+void LpfilterAudioProcessor::releaseResources()
+{
+    // When playback stops, you can use this as an opportunity to free up any
+    // spare memory, etc.
+}
+
+#ifndef JucePlugin_PreferredChannelConfigurations
+bool LpfilterAudioProcessor::isBusesLayoutSupported (const BusesLayout& layouts) const
+{
+    const AudioChannelSet& mainInput = layouts.getMainInputChannelSet();
+    const AudioChannelSet& mainOutput = layouts.getMainOutputChannelSet();
+    
+    // This checks if the input layout matches the output layout
+    if ( mainInput != mainOutput )
+        return false;
+    
+    // Check for only mono or stereo layout
+    if ( mainInput.size() > 2 )
+        return false;
+    
+    return true;
+    
+}
+#endif
+
+void LpfilterAudioProcessor::processBlock (AudioSampleBuffer& buffer, MidiBuffer& midiMessages)
+{
+    ScopedNoDenormals noDenormals;
+    const int totalNumInputChannels  = getTotalNumInputChannels();
+    const int totalNumOutputChannels = getTotalNumOutputChannels();
+
+    // In case we have more outputs than inputs, this code clears any output
+    // channels that didn't contain input data, (because these aren't
+    // guaranteed to be empty - they may contain garbage).
+    // This is here to avoid people getting screaming feedback
+    // when they first compile a plugin, but obviously you don't need to keep
+    // this code if your algorithm always overwrites all the output channels.
+    for (int i = totalNumInputChannels; i < totalNumOutputChannels; ++i)
+        buffer.clear (i, 0, buffer.getNumSamples());
+
+    // Apply gain
+    for (int channel = 0; channel < totalNumInputChannels; ++channel)
+    {
+        buffer.applyGain(*gain);
+    }
+}
+
+//==============================================================================
+bool LpfilterAudioProcessor::hasEditor() const
+{
+    return false; // (change this to false if you choose to not supply an editor)
```

Comment:
I tried to load the plugin (AU format) in Ableton Live and it crashes.

If `createEditor` method returned a nullptr then `hasEditor` should return false.
So, change the return value to 'true'.